betty/project/extension/raspberry_mint/config.py

```diff
@@ -4,22 +4,96 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from collections import defaultdict
+from enum import StrEnum
+from typing import TYPE_CHECKING, final
 
 from typing_extensions import override
 
-from betty.assertion import OptionalField, assert_record
+from betty.assertion import OptionalField, assert_len, assert_mapping, assert_record
 from betty.config import Configuration
-from betty.model.config import EntityReference, EntityReferenceSequence
+from betty.data import Key
+from betty.exception import HumanFacingExceptionGroup
+from betty.plugin.config import (
+    PluginInstanceConfiguration,
+    PluginInstanceConfigurationSequence,
+)
 from betty.project.extension._theme import ColorConfiguration
 
 if TYPE_CHECKING:
-    from collections.abc import Iterable, Sequence
+    from collections.abc import Iterable, Mapping, MutableMapping, Sequence
 
+    from betty.content_provider import ContentProvider, ContentProviderDefinition
     from betty.mutability import Mutable
     from betty.serde.dump import Dump, DumpMapping
 
 
+@final
+class Region(StrEnum):
+    """
+    A theme region in which content can be placed.
+    """
+
+    FRONT_PAGE = "front"
+
+
+@final
+class RegionalContentConfiguration(Configuration):
```

Comment:
Can we make this available to any third party code? E.g. move it out of Raspberry Mint